experimental/framework/fluid-static/src/signalManager.ts

```diff
@@ -182,3 +186,84 @@ export class SignalManager extends TypedEventEmitter<IErrorEvent> implements ISi
         this.submitSignal(broadcastSignalName, payload);
     }
 }
+
+/**
+ * Note: currently experiemental and under development
+ *
+ * DataObject implementation of ISignaler for fluid-static plug-and-play.  Allows fluid-static
+ * users to get an ISignaler without a custom DO.  Where possible, consumers should instead
+ * create a Signaler themselves instead of using the DO wrapper to avoid the DO overhead.
+ */
+// eslint-disable-next-line @typescript-eslint/ban-types
+export class SignalManager extends DataObject<{}, undefined, IErrorEvent> implements EventEmitter, ISignaler {
```

Comment:
Just for some clarification on the different classes here, is the following a correct understanding?

- IRuntimeSignaler - Interface that has a subset of `IRuntime` that is required for signaling
- ISignaler - Defines the interface that will be used to interact with the signals emitted by runtime
- Signaler - Class that takes in runtime (or anything else that matches IRuntimeSignaler) that provides the `ISignaler` interface for consumers to use
- SignalManager - Gives an example of how to set up a data object with a `Signaler` instance that is provided the runtime and then acts as a pass-through for all of the events it emits

So developers can either choose to directly use the `SignalManager` object in their container schema and pass it to their application. Or if they want to go about using signals within their own data object, they can similarly set up a `Signaler` instance and work with it directly.

Reply:
yup that's exactly right.  hoping it's not too confusing to understand 😵